external/js/cothority/src/bevm/bevm-instance.ts

```diff
@@ -238,27 +205,27 @@ export class BEvmClient extends Instance {
      */
     static async fromByzcoin(bc: ByzCoinRPC, iid: InstanceID,
                              waitMatch: number = 0,
-                             interval: number = 1000): Promise<BEvmClient> {
+                             interval: number = 1000): Promise<BEvmInstance> {
         const instance = await Instance.fromByzcoin(bc, iid, waitMatch, interval);
 
-        return new BEvmClient(bc, instance);
+        return new BEvmInstance(bc, instance);
     }
 
-    private bevmService: BEvmService;
+    private bevmRPC: BEvmRPC;
 
     constructor(private byzcoinRPC: ByzCoinRPC, inst: Instance) {
         super(inst);
 
-        if (inst.contractID.toString() !== BEvmClient.contractID) {
-            throw new Error(`mismatch contract name: ${inst.contractID} vs ${BEvmClient.contractID}`);
+        if (inst.contractID.toString() !== BEvmInstance.contractID) {
+            throw new Error(`mismatch contract name: ${inst.contractID} vs ${BEvmInstance.contractID}`);
         }
     }
 
     /**
      * Set the BEvm service to use
      */
-    setBEvmService(bevmService: BEvmService) {
-        this.bevmService = bevmService;
+    setBEvmRPC(bevmRPC: BEvmRPC) {
```

Comment:
if we suppose that each node has the bevm-service, is this still needed?

Reply:
True, we probably don't need to set this from the outside.

Removed `setBEvmRPC()` and initialize the service in `BEvmInstance` constructor.